### PR TITLE
GKE stateful blueprints

### PR DESCRIPTION
**Pull request by juliocc**
This PR merges 4 bluprints showing how to use GKE Autopilot cluster to do batch processing and HA stateful workloads

- Base Autopilot Cluster blueprint showing how to deploy a secure-by-default cluster following Google's best recommendations
- Kueue on GKE
- Redis Cluster on GKE
- Kafka on GKE

### CODE REVIEW

File: blueprints/gke/patterns/autopilot-cluster/README.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+- Configure cost controls: TBD
+- Configure billing alerts: out of scope for this blueprint
+
+
+<!-- ## Usage Examples -->
+
+<!-- ### Existing cluster with local fleet -->
+
+<!-- ```hcl -->
+<!-- module "jumpstart-0" { -->
```
Comment (apichick) [inline blueprints/gke/patterns/autopilot-cluster/README.md:95]: region must specified, shouldn't it? In all the examples

File: blueprints/gke/patterns/batch/create_jobs.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash 
+
+# Copyright 2023 Google LLC
```
Comment (apichick) [inline blueprints/gke/patterns/batch/create_jobs.sh:3]: 2024

File: blueprints/gke/patterns/kafka/main.tf [hunk 3]
```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2023 Google LLC
```
Comment (apichick) [inline blueprints/gke/patterns/kafka/main.tf:2]: 2024

File: blueprints/gke/patterns/kafka/outputs.tf [hunk 4]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Copyright 2023 Google LLC
```
Comment (apichick) [inline blueprints/gke/patterns/kafka/outputs.tf:2]: 2024

File: blueprints/gke/patterns/kafka/versions_override.tf
Comment (apichick) [inline blueprints/gke/patterns/kafka/versions_override.tf:1]: Why is this file required?

File: blueprints/gke/patterns/redis-cluster/main.tf [hunk 5]
```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright 2023 Google LLC
```
Comment (apichick) [inline blueprints/gke/patterns/redis-cluster/main.tf:2]: 2024

File: blueprints/gke/patterns/batch/manifest-templates/cluster-resources/resource-flavor.yaml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2023 Google LLC
```
Comment (apichick) [inline blueprints/gke/patterns/batch/manifest-templates/cluster-resources/resource-flavor.yaml:1]: 2024

File: blueprints/gke/patterns/batch/providers.tf [hunk 7]
```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2023 Google LLC
```
Comment (apichick) [inline blueprints/gke/patterns/batch/providers.tf:2]: 2024

File: blueprints/gke/patterns/redis-cluster/providers.tf [hunk 8]
```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2023 Google LLC
```
Comment (apichick) [inline blueprints/gke/patterns/redis-cluster/providers.tf:2]: 2024

File: blueprints/gke/patterns/autopilot-cluster/README.md (hunk 1 above)
Comment (juliocc) [reply blueprints/gke/patterns/autopilot-cluster/README.md:95]: Yes, but we hve to re-write all those examples with the new variables. I'll do that in a separate PR

File: blueprints/gke/patterns/redis-cluster/main.tf (hunk 5 above)
Comment (juliocc) [reply blueprints/gke/patterns/redis-cluster/main.tf]: because of this, couldn't find any other way gavinbunney/terraform-provider-kubectl#226

File: blueprints/gke/patterns/kafka/versions_override.tf
Comment (juliocc) [reply blueprints/gke/patterns/kafka/versions_override.tf:1]: because of this, couldn't find any other way https://github.com/gavinbunney/terraform-provider-kubectl/issues/226

**Review by apichick — APPROVED** [review]
